testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestAvroSchemaEvolution.java

```diff
@@ -0,0 +1,283 @@
+                        ")",
+                tableName,
+                ORIGINAL_SCHEMA));
+        insertData(tableName, 0, "'stringA0'", "0");
+    }
+
+    @AfterTestWithContext
+    public void dropTestTable()
+    {
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+    }
+
+    @Test(groups = AVRO)
+    public void testSelectTable()
+    {
+        assertThat(query(format("SELECT string_col FROM %s", tableName)))
+                .containsExactlyInOrder(row("stringA0"));
+    }
+
+    @Test(groups = AVRO)
+    public void testCreateTableLike()
+    {
+        String createTableLikeName = "test_avro_like";
+        query(format(
+                "CREATE TABLE %s (LIKE %s INCLUDING PROPERTIES)",
+                createTableLikeName,
+                tableName));
+
+        insertData(createTableLikeName, 0, "'stringA0'", "0");
+
+        assertThat(query(format("SELECT string_col FROM %s", createTableLikeName)))
+                .containsExactlyInOrder(row("stringA0"));
+        query("DROP TABLE IF EXISTS " + createTableLikeName);
+    }
+
+    @Test(groups = AVRO)
+    public void testInsertAfterSchemaEvolution()
+    {
+        assertThat(query(selectStarStatement))
+                .containsOnly(
+                        createRow(0, "stringA0", 0));
+
+        alterTableSchemaTo(ADDED_COLUMN_SCHEMA);
+        insertData(tableName, 1, "'stringA1'", "1", "101");
+        assertThat(query(selectStarStatement))
+                .containsOnly(
+                        createRow(0, "stringA0", 0, 100),
+                        createRow(1, "stringA1", 1, 101));
+    }
+
+    @Test(groups = AVRO)
+    public void testSchemaEvolutionWithIncompatibleType()
+    {
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("string_col", "varchar", "", ""),
+                                row("int_col", "integer", "", "")));
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        createRow(0, "stringA0", 0));
+
+        alterTableSchemaTo(INCOMPATIBLE_TYPE_SCHEMA);
+        QueryAssert.assertQueryFailure(() -> query(selectStarStatement))
+                .hasMessageContaining("Found int, expecting string");
+    }
+
+    @Test(groups = AVRO)
+    public void testSchemaEvolution()
+    {
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("string_col", "varchar", "", ""),
+                                row("int_col", "integer", "", "")));
+
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        createRow(0, "stringA0", 0));
```

Comment:
Can we inline.

Reply:
And other places too.

Reply:
Same as here https://github.com/trinodb/trino/pull/9370#discussion_r721163357